### DISCUSSION TITLE
Insert html

### DISCUSSION THREAD

**Discussion by pajaro1966** [marp-team/marp#349] [Q&A] [answered]
I would like to insert html code and see it rendered in Marp. Is it possible?
The html code show texts with interlinear translations like this:

![image](https://user-images.githubusercontent.com/3808217/189519646-8c8ee424-b116-4c74-b726-dd04c242b59d.png)

In Markdown files I can usually add html code and it's rendered nicely, but I cannot reproduce this in Marp.
If possible I would like to use CSS in order to control the code rendition.

This is the code:

```
<hr/>
<div style="margin-top:1px; font-family: Times; font-size:26px; line-height: 20px;">
<span style="font-size: 100% ; display: inline-block; text-align:center; padding-bottom: 15px;font-family: Times; margin-right: 5px;">
<span style="padding-bottom: 2px; ; display: block; padding-bottom: 10px;border-width: 3px;">με</span>
<span style="font-size: 75%; color:grey;border: solid; border-width:1px; border-padding:4px; border-radius: 5px; display: block; padding: 3px;">me</span></span>
<span style="font-size: 100% ; display: inline-block; text-align:center; padding-bottom: 15px;font-family: Times; margin-right: 5px;">
<span style="padding-bottom: 2px; ; display: block; padding-bottom: 10px;border-width: 3px;">λένε</span>
<span style="font-size: 75%; color:grey;border: solid; border-width:1px; border-padding:4px; border-radius: 5px; display: block; padding: 3px;">dicen</span></span>
<span style="font-size: 100% ; display: inline-block; text-align:center; padding-bottom: 15px;font-family: Times; margin-right: 5px;">
<span style="padding-bottom: 2px; ; display: block; padding-bottom: 10px;border-width: 3px;">Αντόνιο</span>
<span style="font-size: 75%; color:grey;border: solid; border-width:1px; border-padding:4px; border-radius: 5px; display: block; padding: 3px;">Antonio</span></span>
<span style="font-size: 100% ; display: inline-block; text-align:center; padding-bottom: 15px;font-family: Times; margin-right: 5px;">
<span style="padding-bottom: 2px; ; display: block; padding-bottom: 10px;border-width: 3px;">Ρεβουέλτα</span>
<span style="font-size: 75%; color:grey;border: solid; border-width:1px; border-padding:4px; border-radius: 5px; display: block; padding: 3px;">Revuelta</span></span>
<span style="font-size: 100% ; display: inline-block; text-align:center; padding-bottom: 15px;font-family: Times; margin-right: 5px;">
<span style="padding-bottom: 2px; ; display: block; padding-bottom: 10px;border-width: 3px;"></span></span></div><hr/ style="display: block; margin-top: 0.5em;  margin-bottom: 0.5em;  margin-left: auto;  margin-right: auto;  border-style: inset; border-width: 1px;">
<div style="margin-top:1px; font-family: Times; font-size:26px; line-height: 20px;">
<span style="font-size: 100% ; display: inline-block; text-align:center; padding-bottom: 15px;font-family: Times; margin-right: 5px;">
<span style="padding-bottom: 2px; ; display: block; padding-bottom: 10px;border-width: 3px;"></span></span></div>
```
Thanks for your help-

**Comment by yhatt** [comment, marked as answer]
Marp ecosystem is secure first so rendering the most of HTML tags is disabled by default. You can enable rendering of raw HTML tags from options. 

### [Marp CLI](https://github.com/marp-team/marp-cli#options:~:text=Marpit%20based%20engine-,html,-boolean%20%7C%20object)

Add **`--html`** option.

```bash
marp --html your-slide.md
```

### [Marp for VS Code](https://github.com/marp-team/marp-vscode#enable-html-in-marp-markdown-%EF%B8%8F)

Open [VS Code settings](https://code.visualstudio.com/docs/getstarted/settings),  and turn on "**Markdown > Marp: Enable HTML**" (`markdown.marp.enableHtml`).

**Reply by lucastho** [reply]
Is there similar for marp-core or marpit?

**Reply by yhatt** [reply]
In [Marp Core](https://github.com/marp-team/marp-core), you can set through the `html` constructor option. Check out [Marp Core documentation](https://github.com/marp-team/marp-core#constructor-options).

```javascript
import { Marp } from '@marp-team/marp-core'

const marp = new Marp({
  html: true
});
```

[Marpit framework](https://marpit.marp.app) is delegating to [markdown-it](https://github.com/markdown-it/markdown-it) instance whether enabled HTML. By default, [markdown-it instance](https://markdown-it.github.io/markdown-it/#MarkdownIt.new) will be initialized as [CommonMark compliant mode](https://github.com/markdown-it/markdown-it/blob/master/lib/presets/commonmark.mjs), and _enabled HTML by default_.

```javascript
import { Marpit } from '@marp-team/marpit'

// Marpit instance enables HTML by default
const marpit = new Marpit();
```

To set the markdown-it constructor option in Marpit constructor explicitly, pass the option object to `markdown` field of Marpit constructor. [Refer to the API documentation](https://marpit-api.marp.app/marpit#:~:text=in%20current%20instance.-,markdown,-MarkdownIt%20%7C%20string).

```javascript
import { Marpit } from '@marp-team/marpit'

const marpit = new Marpit({
  // Set the markdown-it constructor options explicitly
  markdown: {
    html: true,
  },
});
```

**Comment by pajaro1966** [comment]
Thank you so much! It works like a charm! Sorry for asking this. It's my first hour using Marp and, although I could do. almost everything (it's easy for beginners), I was clueless about the insertion of html.
Another question, how could I apply a css general style to all my html examples within a singles presentation? Is it possible to put css code in a Marp file and make it operate over certain html chunks? 
Thanks in advance for your help. I really appreciate.
Antonio

**Reply by yhatt** [reply]
https://marpit.marp.app/theme-css?id=tweak-style-through-markdown would be helpful.

> ⚠️ Please avoid to ask multiple questions in a single thread. Each Q&A threads should have a single theme [(the marked answer is correspond to the thread one-to-one)](https://docs.github.com/ja/discussions/collaborating-with-your-community-using-discussions/participating-in-a-discussion#marking-a-comment-as-an-answer).
>
> If there is another question in a sub thread (comment), users who met the same trouble would be difficult to discover the answered Q&A.
>
> - [More than one question per post - Meta Stack Overflow](https://meta.stackoverflow.com/a/275909)
> - [How To Ask Good Questions On Technical And Scientific Forums](https://www.biostars.org/p/75548/#:~:text=Avoid%20asking%20multiple%20questions%20in%20the%20same%20post)
> - [How to write a quality question on Microsoft Q&A | Microsoft Docs](https://docs.microsoft.com/en-us/answers/support/quality-question#ask-one-question-at-a-time)
